prov/gni/test/rdm_sr.c

```diff
@@ -736,6 +737,54 @@ Test(rdm_sr, inject)
 	rdm_sr_xfer_for_each_size(do_inject, 1, INJECT_SIZE);
 }
 
+/*
+ * this test attempts to demonstrate issue ofi-cray/libfabric-cray#559.
+ * For domains with control_progress AUTO, this test should not hang.
+ */
+Test(rdm_sr, inject_progress)
+{
+	int ret, len = 64;
+	ssize_t sz;
+	struct fi_cq_tagged_entry cqe;
+	uint64_t s[NUMEPS] = {0}, r[NUMEPS] = {0}, s_e[NUMEPS] = {0};
+	uint64_t r_e[NUMEPS] = {0};
+
+	rdm_sr_init_data(source, len, 0x23);
+	rdm_sr_init_data(target, len, 0);
+
+	sz = fi_inject(ep[0], source, len, gni_addr[1]);
+	cr_assert_eq(sz, 0);
+
+	sz = fi_recv(ep[1], target, len, rem_mr[1], gni_addr[0], source);
+	cr_assert_eq(sz, 0);
+
+	/*
+	 * do progress until send counter is updated.
+	 * This works because we have FI_PROGRESS_AUTO for control progress
+	 */
+	while (fi_cntr_read(send_cntr[0]) < 1) {
+		pthread_yield();
+	}
```

Comment:
It should fail since progessing the send_cntr[0] doesn't progress the domain[1]/ep[1] connection setup if there is no control progress thread for domain[1].    I'll check that.